src/lib/fcitx-utils/log.cpp

```diff
@@ -23,7 +23,10 @@ namespace {
 
 FCITX_DEFINE_LOG_CATEGORY(defaultCategory, "default");
 
+using log_rules_t = std::vector<std::pair<std::string, LogLevel>>;
```

Comment:
Please use CamelCase for type name.

Also, I suggest you create a alias for std::pair<std::string, LogLevel>, not the vector type. It would be more readable for knowing that this is a vector.